src/v/storage/index_state.cc

```diff
@@ -135,6 +135,7 @@ void index_state::serde_write(iobuf& out) const {
     write(tmp, position_index.copy());
     write(tmp, batch_timestamps_are_monotonic);
     write(tmp, with_offset);
+    write(tmp, non_data_timestamps);
```

Comment:
@VladLazar did this not need a bump to the serde version?

Reply:
It was bumped in https://github.com/redpanda-data/redpanda/pull/8509 for which this PR was a follow up to.

Reply:
thanks!